### PR TITLE
feat: added support for s3 and fs config file provider

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         |✖                                                                        |
| New feature     | ✔                                                                   |
| Breaking change |✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                       |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: src/common/providers/FSProvider.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { IFileProvider } from '../interfaces';
```
Comment (asafMasa) [inline src/common/providers/FSProvider.ts:1]: rename file name to camelCase

File: src/common/providers/S3Provider.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
 /* eslint-disable @typescript-eslint/naming-convention */
 import { promises } from 'fs';
```
Comment (asafMasa) [inline src/common/providers/S3Provider.ts:2]: rename file name to camel case s3Provider.ts  - check for all files

File: config/test.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "port": "8080"
   },
   "mapproxy": {
+    "fileProvider": "fs",
```
Comment (asafMasa) [inline config/test.json:15]: open issue to update the helm chart

**Review by asafMasa — CHANGES_REQUESTED** [review]
good job

File: src/common/enums/Providers.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export enum Providers {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/common/enums/Providers.ts]: fixed in a45322b

File: src/common/providers/FSProvider.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/providers/FSProvider.ts]: fixed in a45322b

File: src/common/providers/S3Provider.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/providers/S3Provider.ts]: fixed in a45322b

File: config/test.json (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply config/test.json:15]: fixed: https://rnd-hub.atlassian.net/browse/MAPCO-439

**Review by asafMasa — APPROVED** [review]

**Pull request merged**